### PR TITLE
Undefined Map snowpack

### PR DESCRIPTION
**Pull request by Aslemammad**

### CODE REVIEW

**Comment by Aslemammad** [comment]
This fix didn't work; I'm working on the new fix.

**Comment by Aslemammad** [comment]
The problem is the `new Map([...Value, ...Value2])` syntax, which when it gets transpiled, the `concat` method mess the values.

File: src/core/vanilla.ts [hunk 1]
```diff
@@ -15,6 +15,14 @@ const hasInitialValue = <T extends Atom<unknown>>(
...
   'init' in atom
 
+const concatMap = <K, V>(m1: Map<K, V>, m2: Map<K, V>): Map<K, V> => {
+  // Map([...m1, ...m2]) alternative
+  const newMap = new Map(m1)
+  m2.forEach((value, key) => {
+    newMap.set(key, value)
+  })
+  return newMap
+}
```
Comment (dai-shi) [inline src/core/vanilla.ts:25]:
```suggestion
}

```

File: src/core/vanilla.ts [hunk 2]
```diff
@@ -287,7 +296,7 @@ const readAtomState = <Value>(
           state.u((prev) =>
             setAtomValue(
               state,
-              new Map([...prev, ...wip]),
+              concatMap(prev, wip),
```
Comment (dai-shi) [inline src/core/vanilla.ts:299]: Same one in L315

**Comment by Aslemammad** [comment]
@dai-shi #327 fixed now. [codesandbox](https://codesandbox.io/s/infallible-cache-td09c?file=/package.json).

**Review by dai-shi — APPROVED** [review]

**Pull request merged**